embed_images/shot_boundary_detection.py

```diff
@@ -0,0 +1,105 @@
+# /// script
+# description = "Summarize podcasts"
+# requires-python = ">=3.10, <3.13"
+# dependencies = ["daft", "pandas","numpy", "transformers","torchvision","matplotlib","av","yt-dlp"]
+# ///
+import daft
+from daft.functions import embed_image
+from daft import col, Window, DataType as dt
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+@daft.func(return_dtype=dt.float32())
+def l2_distance(a: np.ndarray | None, b: np.ndarray | None) -> float:
```

Comment:
no action needed, but this would probably be something nice to have as a native rust expression.